src/Std/Classes/Ord.lean

```diff
@@ -276,6 +276,28 @@ theorem compare_beq_iff_eq {a b : α} : cmp a b == .eq ↔ a = b :=
 
 end LawfulEq
 
+section LawfulBEq
+
+/--
+A typeclass for comparison functions satisfying `cmp a b = .eq` if and only if the boolean equality
+`a == b` holds.
+-/
+class LawfulBEqCmp {α : Type u} [BEq α] (cmp : α → α → Ordering) : Prop where
```

Comment:
Might be nice to note in the docstring the relationship to the very similarly named `LawfulEqCmp`.